reference-implementation/lib/resolver.js

```diff
@@ -1,100 +1,60 @@
 'use strict';
 const { URL } = require('url');
 const assert = require('assert');
-const { tryURLLikeSpecifierParse, BUILT_IN_MODULE_SCHEME, BUILT_IN_MODULE_PROTOCOL } = require('./utils.js');
+const {
+  tryURLLikeSpecifierParse,
+  BUILT_IN_MODULE_SCHEME
+} = require('./utils.js');
 
 // TODO: clean up by allowing caller (and thus tests) to choose the list of built-ins?
 const supportedBuiltInModules = new Set([
   `${BUILT_IN_MODULE_SCHEME}:blank`,
   `${BUILT_IN_MODULE_SCHEME}:blank/for-testing` // NOTE: not in the spec.
 ]);
 
-exports.resolve = (specifier, parsedImportMap, scriptURL) => {
-  const asURL = tryURLLikeSpecifierParse(specifier, scriptURL);
-  const normalizedSpecifier = asURL ? asURL.href : specifier;
-  const scriptURLString = scriptURL.href;
-
-  for (const [scopePrefix, scopeImports] of Object.entries(parsedImportMap.scopes)) {
-    if (scopePrefix === scriptURLString ||
-        (scopePrefix.endsWith('/') && scriptURLString.startsWith(scopePrefix))) {
-      const scopeImportsMatch = resolveImportsMatch(normalizedSpecifier, scopeImports);
-      if (scopeImportsMatch !== null) {
-        return scopeImportsMatch;
-      }
-    }
-  }
-
-  const topLevelImportsMatch = resolveImportsMatch(normalizedSpecifier, parsedImportMap.imports);
-  if (topLevelImportsMatch !== null) {
-    return topLevelImportsMatch;
+exports.resolve = (specifier, parsedImportMap, scriptURLparameter) => {
```

Comment:
If it's easy it would be nice to have this take a `URL` as its third argument, directly. (Bad JavaScript API design, but good for a reference implementation mirroring the spec.) I kind of thought it did previously...